src/Rules/PhpDoc/IncompatibleRequireImplementsTypeRule.php

```diff
@@ -0,0 +1,122 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\PhpDoc;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassLike;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Rules\ClassCaseSensitivityCheck;
+use PHPStan\Rules\ClassNameNodePair;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\Generic\GenericObjectType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeTraverser;
+use PHPStan\Type\VerbosityLevel;
+use function array_merge;
+use function count;
+use function sprintf;
+
+/**
+ * @implements Rule<ClassLike>
+ */
+class IncompatibleRequireImplementsTypeRule implements Rule
+{
+
+	public function __construct(
+		private ReflectionProvider $reflectionProvider,
+		private ClassCaseSensitivityCheck $classCaseSensitivityCheck,
+		private UnresolvableTypeHelper $unresolvableTypeHelper,
+		private bool $checkClassCaseSensitivity,
+	)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return ClassLike::class;
```

Comment:
but it doesn't work for trait defintions. it would not error on 

```
/**
 * @phpstan-require-implements SomeTrait
 */
trait InvalidTrait1 {}
```

Reply:
Two separate rules then please.